drivers/spi/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ zephyr_library()
 
 zephyr_library_sources_ifdef(CONFIG_SPI_DW		spi_dw.c)
 zephyr_library_sources_ifdef(CONFIG_SPI_INTEL		spi_intel.c)
-zephyr_library_sources_codegen_ifdef(CONFIG_SPI_STM32	spi_ll_stm32.c.jinja2 SEARCH_PATH ${PROJECT_SOURCE_DIR}"/templates/")
+zephyr_library_sources_codegen_ifdef(CONFIG_SPI_STM32	spi_ll_stm32_device_template.h CODEGEN_OUTPUT spi_ll_stm32_device.h SEARCH_PATH ${PROJECT_SOURCE_DIR}"/templates/")
```

Comment:
Can you provide a more easy to use macro?
Ideally one should only call zephyr_library_sources_jinja2_ifdef(CONFIG_SPI_STM32	 spi_ll_stm32.c)
Then jinja2 should generate spi_ll_stm32_devices.c/h from  spi_ll_stm32_devices.c.jinja

Reply:
well, you can omit CODEGEN_OUTPUT and SEARCH_PATH if you don't need them

CODEGEN_OUTPUT is to control the name of the rendered file which can be nice in many cases

SEARCH_PATH is set "include_paths", so if you have templates in other folders that you want to reuse in your template (with the include/import or extends keywords)